### ISSUE TITLE
Error with nushell 0.83.0

### ISSUE THREAD

**Issue by 18183883296** [ajeetdsouza/zoxide#595]
The project is not well-adapted to the latest version of nushell, version 0.83.0, and it displays the following error:
```
Error: nu::parser::input_type_mismatch

  × Command does not support string input.
    ╭─[C:\Users\Sunrise\.zoxide.nu:27:1]
 27 │   let arg0 = ($rest | append '~').0
 28 │   let path = if (($rest | length) <= 1) and ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {
    ·                           ───┬──
    ·                              ╰── command doesn't support string input
 29 │     $arg0
    ╰────
```

**Comment by asim215** [comment]
Solution is following:
`($rest | str length).0 <= 1`
I also changed everywhere:
from `let-env x` to `$env.x`

```
# Initialize hook to add new entries to the database.
if (not ($env | default false __zoxide_hooked | get __zoxide_hooked)) {
  $env.__zoxide_hooked = true
  $env.config = ($env | default {} config).config
  $env.config = ($env.config | default {} hooks)
  $env.config = ($env.config | update hooks ($env.config.hooks | default {} env_change))
  $env.config = ($env.config | update hooks.env_change ($env.config.hooks.env_change | default [] PWD))
  $env.config = ($env.config | update hooks.env_change.PWD ($env.config.hooks.env_change.PWD | append {|_, dir|
    zoxide add -- $dir
  }))
}

# =============================================================================
#
# When using zoxide with --no-cmd, alias these internal functions as desired.
#

# Jump to a directory using only keywords.
def-env __zoxide_z [...rest:string] {
  let arg0 = ($rest | append '~').0
  let path = if (($rest | str length).0 <= 1) and ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {
    $arg0
  } else {
    (zoxide query --exclude $env.PWD -- $rest | str trim -r -c "\n")
  }
  cd $path
}

# Jump to a directory using interactive search.
def-env __zoxide_zi  [...rest:string] {
  cd $'(zoxide query --interactive -- $rest | str trim -r -c "\n")'
}

# =============================================================================
#
# Commands for zoxide. Disable these using --no-cmd.
#

alias z = __zoxide_z
alias zi = __zoxide_zi
```

**Comment by mati865** [comment]
@asim215  this is not a fix because it changes the way zoxide works.
Original code checks if array `rest` has 1 or less items, your code checks if first element of the array is 1 character long or shorter.

**Comment by asim215** [comment]
For now I just removed this check in if statement. So it becomes:
` let path = if ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {`

**Comment by amtoine** [comment]
as said in https://github.com/ajeetdsouza/zoxide/pull/596, this issue has been fixed in https://github.com/nushell/nushell/pull/9816 and should be released in a patch release soon :relieved:

**Comment by asim215** [comment]
It looks like original if check works fine now with updated nushell 0.83.1.
`  let path = if (($rest | length) <= 1) and ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {
` 
Thanks!

**Comment by ajeetdsouza** [comment]
Awesome, closing this then. Thanks!

**Comment by charlesroper** [comment]
I've updated to nushell 0.83.1 and still seeing the error related to `let-env`

![2023-07-31_08](https://github.com/ajeetdsouza/zoxide/assets/176/b5f8af2d-60fc-4cec-8a74-1b0e80022d38)

I thought maybe I could just edit `.zoxide.nu` but it says "Code generated by zoxide. DO NOT EDIT." at the top. If I manually replace all `let-env` instances to `$env.`, these changes get replaced by the init script in `env.nu`.

Solution was to disable to init script from running every time.

I guess if the `let-env`/`$env.` change remains in nu, then the init script will need updating.

**Comment by Kissaki** [comment]
@charlesroper yes, the [nushell 0.83.1 release notes explicitly note that let-env won't return](https://www.nushell.sh/blog/2023-07-30-nushell_0_83_1.html#bring-the-let-env-command-back-to-life)

That one is pretty simple to update though with a simple replace/update.

**Comment by JonasLeonhard** [comment]
@charlesroper 
I ran into the same issue. My `zoxide init nushell` contained 'let-env', even though i had the newest zoxide version (v0.9.1) installed via `brew install zoxide`. 
The file for the template should have been fixed in the main branch: https://github.com/ajeetdsouza/zoxide/blob/main/templates/nushell.txt . 
To fix `zoxide init nushell` to contain the updated script, i installed zoxide via `cargo install --git https://github.com/ajeetdsouza/zoxide`. 
Hopefully that helps.

**Comment by JonasLeonhard** [comment]
@ajeetdsouza 
Is that something that needs to be adressed? It seems like to me that the brew install version contains old code.

**Comment by ajeetdsouza** [comment]
Released in 0.9.2.

**Comment by mikecrowe-8451** [comment]
For anybody else coming here, make sure `nushell` is at `0.83.1` -- `.0` still fails
